File: coolrmi/hu.qgears.coolrmi/src/hu/qgears/coolrmi/serializer/EnumSerializer.java
package hu.qgears.coolrmi.serializer;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

class EnumSerializer extends TypeSerializer {
	public EnumSerializer() {
		super(TypeId.Enum, null, null);
	}

	@Override
	public boolean canSerializeIsSpecial() {
		return true;
	}

	@Override
	public boolean canSerialize(PortableSerializer serializer, JavaType typ) {
		return typ.getCls().isEnum();
	}

	@Override
	public void writeType(PortableSerializer serializer, OutputStream os,
			JavaType typ) throws IOException {
		super.writeType(serializer, os, typ);
		Utils.writeString(os, serializer.getPortableClassName(typ));
	}

	@Override
	public JavaType readType(PortableSerializer serializer, InputStream is)
			throws IOException, ClassNotFoundException {
		return new JavaType(serializer.loadClass(Utils.readString(is)));
	}

	@Override
	public void serialize(PortableSerializer serializer, OutputStream os,
			Object o, JavaType typ) throws IOException {
		Utils.writeString(os, ((Enum<?>) o).name());
	}

	@SuppressWarnings({ "unchecked", "rawtypes" })
	@Override
	public Object deserialize(PortableSerializer serializer, InputStream is,
			JavaType typ) throws IOException {
		String value = Utils.readString(is);
		return Enum.valueOf((Class) typ.getCls(), value);
	}

}

File: coolrmi/CoolRMI.Net/Serializer/EnumSerializer.cs
using System;
using System.IO;

namespace CoolRMI.Net.Serializer
{
    public class EnumSerializer : TypeSerializer
    {
        public EnumSerializer() : base(TypeId.Enum, null) {}

        public override bool CanSerializeIsSpecial => true;

        public override bool CanSerialize(PortableSerializer serializer,
            Type typ)
        {
            return typ.IsEnum;
        }

        public override void WriteType(PortableSerializer serializer,
            BinaryWriter bw, Type typ)
        {
            base.WriteType(serializer, bw, typ);
            serializer.WriteClassName(bw, typ);
        }

        public override Type ReadType(PortableSerializer serializer,
            BinaryReader br)
        {
            return serializer.ReadClassName(br);
        }

        public override void Serialize(PortableSerializer serializer,
            BinaryWriter bw, object o, Type typ)
        {
            bw.WritePString(Enum.GetName(typ, o));
        }

        public override object Deserialize(PortableSerializer serializer,
            BinaryReader br, Type typ)
        {
            var names = typ.GetEnumNames();
            var vals = typ.GetEnumValues();
            var i = Array.IndexOf(names, br.ReadPString());
            return vals.GetValue(i);
        }
    }
}
